.gitlab-ci.yml

```diff
@@ -22,65 +28,55 @@ include:
   - project: 'Northern.tech/Mender/mendertesting'
     file: '.gitlab-ci-github-status-updates.yml'
 
-test:unit:
-  stage: test
-  before_script:
-    - apt-get -qq update && apt-get install -yq $(cat deb-requirements.txt)
-    # start the dbus service
-    - service dbus start
-
-build:make:
+build:docker:
+  tags:
+    - mender-qa-worker-generic-light
   stage: build
   needs: []
-  image: golang:1.17-alpine3.14
+  variables:
+    DOCKER_BUILDKIT: 1
+  image: "docker:${DOCKER_VERSION}"
+  services:
+    - "docker:${DOCKER_VERSION}-dind"
   before_script:
-    - apk add --update git make gcc pkgconfig libc-dev glib-dev
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
   script:
-    - make build
-
-generate-qa-trigger:
-  image: python:alpine
-  stage: trigger_prep
-  rules:
-    - if: '$CI_COMMIT_BRANCH =~ /^(master|[0-9]+\.[0-9]+\.x)$/'
-    - if: '$CI_PIPELINE_SOURCE == "pipeline"'
-      when: never
-    - if: '$CI_PIPELINE_SOURCE == "schedule"'
-      when: never
-  before_script:
-    - apk add --no-cache git
-    - pip3 install pyyaml
-    - wget -q https://raw.githubusercontent.com/mendersoftware/mender-qa/master/scripts/generate_client_publish_job.py
-  script:
-    - python generate_client_publish_job.py --trigger ${CI_PROJECT_NAME} --version ${CI_COMMIT_REF_NAME} --filename gitlab-ci-client-qemu-publish-job.yml
+    - echo "building ${CI_PROJECT_NAME} for ${DOCKER_BUILD_SERVICE_IMAGE}"
+    - docker context create builder
+    - docker buildx create builder --use --driver-opt network=host --buildkitd-flags '--debug --allow-insecure-entitlement network.host' --platform "$MULTIPLATFORM_PLATFORMS"
+    - |-
+      docker buildx build \
+      --file ${DOCKER_DIR:-.}/${DOCKERFILE:-Dockerfile} \
+      --build-arg GIT_COMMIT_TAG="${DOCKER_PUBLISH_COMMIT_TAG}" \
+      --platform "$MULTIPLATFORM_PLATFORMS" \
+      --provenance false \
+      --output "type=oci,dest=image.oci" \
+      ${DOCKER_DIR:-.}
   artifacts:
     paths:
-      - gitlab-ci-client-qemu-publish-job.yml
+      - image.oci
+    expire_in: '1 week'
 
-trigger:mender-qa:
-  stage: trigger
+publish:docker:
```

Comment:
Same here: you can start from the default job: https://github.com/mendersoftware/mendertesting/blob/master/.gitlab-ci-check-docker-build.yml#L177
```suggestion
# excludes non multiplatform build job
publish:image:
  rules:
    - when: never
```

Reply:
> # excludes non multiplatform build job

Why would we want to build using plain docker?

Reply:
> > `# excludes non multiplatform build job`
> 
> Why would we want to build using plain docker?

I'm not following.. With the [multiplatform build job](https://github.com/mendersoftware/mendertesting/blob/cd67132494005d35fd867bb4bb6d5540f2e3bc08/.gitlab-ci-check-docker-build.yml#L141) we use `docker buildx`, not plain docker.

I'm suggesting the:
```
publish:image:
  rules:
    - when: never
```
because otherwise you get both the `publish:image-multiplatform:` and the `publish:image:` jobs (duplicated jobs)

Reply:
Sorry, I misunderstood. I should read the snippet more carefully.